dcc-portal-ui/app/scripts/pathwayviewer/js/services.js

```diff
@@ -1213,4 +1213,254 @@ angular.module('icgc.pathwayviewer.directives.services', [])
+      var geneCount = 0;
+      var geneOverlapExistsHashUsingDbIds = Object.assign({}, geneOverlapExistsHash);
+
+      if (geneOverlapExistsHash && annotatedHighlights) {
+        _.forEach(annotatedHighlights, function (annotatedHighlight) {
+          if (angular.isDefined(geneOverlapExistsHashUsingDbIds[annotatedHighlight.uniprotId])) {
+            geneCount++;
+
+            _.forEach(annotatedHighlight.dbIds, function (dbID) {
+              // Swap in Reactome keys but maintain the id we use this to determine overlaps in O(1)
+              // later... The dbID is used as a reference to the reactome SVG nodes...
+              geneOverlapExistsHashUsingDbIds[dbID] = {
+                id: annotatedHighlight.uniprotId,
+                geneId: annotatedHighlight.geneId,
+                geneSymbol: annotatedHighlight.geneSymbol
+              };
+            });
+            delete geneOverlapExistsHashUsingDbIds[annotatedHighlight.uniprotId];
+          }
+        });
+      }
+      console.log(geneCount + ' Overlapped genes validated! ');
+      return geneOverlapExistsHashUsingDbIds;
+    };
+
+    _pathwayDataService.getPathwayData = function (geneSetId, geneSetOverlapFilters) {
+      var pathwayData = {};
+      return _pathwayDataService.getGeneSet(geneSetId)
+      .then(function (geneSet) {
+        pathwayData.geneSet = geneSet;
+
+        return $q.all([
+          _pathwayDataService.getPathwayProteinMap(geneSet),
+          _pathwayDataService.getEntitySetId(geneSet),
+          _pathwayDataService.getPathwayXML(geneSet),
+          _pathwayDataService.getPathwayZoom(geneSet),
+          _pathwayDataService.getUIParentPathways(geneSet)
+        ]);
+      })
+      .then(function (results) {
+        var pathwayProteinMap = results[0];
+        var entitySetId = results[1];
+
+        pathwayData.xml = results[2];
+        pathwayData.zooms = results[3];
+        pathwayData.uiParentPathways = results[4];
+
+        return $q.all([
+          _pathwayDataService.getGeneListData(geneSetOverlapFilters),
+          pathwayProteinMap,
+          _pathwayDataService.getDrugs(entitySetId)
+        ]);
+      })
+      .then(function (results) {
+        var geneListData = results[0];
+        var pathwayProteinMap = results[1];
+        var drugs = results[2];
+
+        var mutationHighlights = _pathwayDataService.getMutationHighlights(pathwayProteinMap);
+        var drugHighlights = _pathwayDataService.getDrugHighlights(drugs, pathwayProteinMap);
+
+        return $q.all([
+          _pathwayDataService.getGeneOverlapExistsHash(geneListData),
+          _pathwayDataService.getGeneAnnotatedHighlights({
+            highlights: mutationHighlights,
+            includeAdvQuery: true
+          }),
+          _pathwayDataService.getGeneAnnotatedHighlights({
+            highlights: drugHighlights,
+            includeAdvQuery: false
+          })
+        ]);
+      })
```

Comment:
Minor nitpick and don't need to refactor this, but in the future favor hashes for `$q.all`

```js
$q.all({
  overlapExistsHash: _pathwayDataService.getGeneOverlapExistsHash(geneListData),
  annotatedMutationHighlights: ...,
  annotatedDrugHighlights: ...,
}).then(function (results) {
    pathwayData.overlaps = _pathwayDataService.getGeneOverlapExistsHashUsingDbIds(
      results.overlapExistsHash, _.union(results.annotatedMutationHighlights, results.annotatedDrugHighlights)
    );
})
```

That way you don't have to scroll back up and check "what was results[0] again?"